users/mtdjr/mtdjr.c

```diff
@@ -26,25 +64,40 @@ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
     case QWERTY:
       if (record->event.pressed) {
         set_single_persistent_default_layer(_QWERTY);
+        #ifdef RGBLIGHT_ENABLE
+          rgblight_sethsv (RGB_HUE, 255, 255);
```

Comment:
For this stuff, you may want to use `layer_state_set` instead. 

For instance: https://github.com/drashna/qmk_firmware/blob/drashna_keymaps/users/drashna/rgb_stuff.c#L300-L351

This covers the default layer, as well.  
Also ... you may want to use the `_noeeprom` versions of these functions instead. Otherwise, you may end up writing to the EEPROM a LOT, which isn't really good.

Reply:
I would definitely add the solenoid stuff as a core feature, just not sure how to do that. Point me in the right direction?

Also, I tried moving the RGB stuff into ```layer_state_set_rgb``` as you suggested but the colors don't change with the layer switch and its not being set to the color defined in ```matrix_init_user```. Any ideas?

Reply:
You'd want to use `layer_state_set_user`.  I sue the rgb function, because it's called by the _user function.  

And as for the matrix_init_user.... yeah, that's a known issue.  If you use the regular ("eeprom") version, it will work.  Or ... add a timer and use `matrix_scan_user`...     
The reason for this, is that matrix init is called before the RGB init (which reads the settings from eeprom ... and over ride whatever you set, unless that wrote to the eeprom).

I have a PR that's sitting that will fix that issue, but it hasn't been merged yet.

Reply:
https://docs.qmk.fm/#/custom_quantum_functions?id=example-layer_state_set_-implementation

Reply:
Cool, ```layer_state_set_user ``` worked. I'll just use the eeprom function for the initial set of the color until your PR is merged. Whats the #? I'll subscribe to it so I know when its ready.

Reply:
It's #3113